### ISSUE TITLE
Providing new OIDs - Short/Long-Name empty string

### ISSUE THREAD

**Issue by matthiasballreich** [openssl/openssl#3909]
Hi there,

i'm working on some Extension Stuff which i want to include into the openssl source. For this i need to add some new OIDs which are currently not supported by openssl. I know i can add them dynamically with OBJ_Create(), but if i decide to make a pull request and for the code i have to add the OIDs statically.

But i noticed a mysterious behaviour. The short and long name of the new added OID will not be printed out by using OBJ_nid2sn() and OBJ_nid2ln(). The integer value of the NID will be printed out correctly. And another thing is that i used the whole time the dll-libraries (i'm on windows) of the stable openssl build 1.1.0f, then i cloned the latest openssl dev 1.1.1-dev from the master branch, which has added the support for the X509v3 Extension Admission and there a new OID was added. I compiled the source, but did not linked my application to the new generated dll-libraries - i use the old ones. And there the short and long name is printed correctly. But if i add a new one to objects.txt only an empty string is printed. As described above the int value of the NID, for example 1093, is printed correctly.

I added the OID to the objects.txt in crypto/objects and run these commands to update the depending headers and files:
- perl crypto/objects/objects.pl crypto/objects/objects.txt crypto/objects/obj_mac.num include/openssl/obj_mac.h
- perl crypto/objects/obj_dat.pl include/openssl/obj_mac.h crypto/objects/obj_dat.h
- perl crypto/objects/objxref.pl crypto/objects/obj_mac.num crypto/objects/obj_xref.txt > crypto/objects/obj_xref.h

These commands are working, too and all the files will be created / updated correctly. Alls short and long names, etc. are added right.

Then i compile and build the openssl source on Windows 10 64Bit and build it for 32Bit which works correctly and without any problems.

Then i run my code and try to print out the generated NID as described above and there only an empty string is printed out.

I addressed the issue to the [users mailing list ](http://openssl.6102.n7.nabble.com/OpenSSL-1-1-0-providing-new-OIDs-to-source-code-tt71424.html) and there Richard Levitte told me that i should try to use `OPENSSL_init_crypto(0, NULL); `at start of my main(). I tried it, but it has no effect.

Then i replaced the old used dll-libraries (from build 1.1.0f) and replaced them with the new generated ones from the compile step above. Then i run my code and the short and long names are printed correctly as i expected it.

So maybe this is not an issue? Must i replace the dll-libraries to which i'm linking everytime i compile the source or not? But if not, why has it worked with the old dll-libraries from 1.1.0f and the new added OID Admission of the master branch as described above?

This is the OID i added (but i tried some more fictive OIDs for testing purposes) to objects.txt
`identified-organization 36 8 3 2	: id-commonpki-at-procuration		: Procuration Information`

And this is my main() code:
```
printf(OBJ_nid2sn(NID_id_commonpki_at_procuration));
printf(OBJ_nid2ln(NID_id_commonpki_at_procuration));
```

If more details needed please tell me.

best regards

**Comment by kroeckx** [comment]
I think you need to remove the space in your long name.

**Comment by kroeckx** [comment]
Never mind the comment about the space, I think that should work.

So I guess the issue you have is that you have 2 set of dll files,
1 in the openssl source, 1 in an other project source. And of
course you need to copy them from one place to the other if you
updated anything in openssl.

**Comment by matthiasballreich** [comment]
> So I guess the issue you have is that you have 2 set of dll files, 1 in the openssl source, 1 in an other project source. And of course you need to copy them from one place to the other if you updated anything in openssl.

yeah, but why has it worked with the old dlls and without copying the new one and the build of the master branch with the new Admission Extension? I eman there is no problem to copy the new dlls every time i build the source. If this is the right way okay, but i don't know if it is a MUST or not you know?

**Comment by kroeckx** [comment]
I have no idea what you mean that it used to work. I can't think
of any case where you modify the openssl source and don't need to
copy it for it to work.

**Comment by matthiasballreich** [comment]
you're right. i have to replace the libraries every time i modify the source. Don't know why the one case i described has worked. I tried this one again and it did not work, but when i copy the libraries it does. So i close this issue here, because this is no issue and it was solved, too. 

In conclusion for anyone who maybe will read this is to link to new generated libraries every time you modify the source and rebuild it.

thanks
